libraries/AP_Proximity/AP_Proximity_LightWareSF45B.cpp

```diff
@@ -0,0 +1,204 @@
+        if (_msg.payload_recv == sizeof(uint32_t)) {
+            _sensor_state.stream_data_type = buff_to_uint32(_msg.payload[0], _msg.payload[1], _msg.payload[2], _msg.payload[3]);
+        }
+        break;
+
+    case MessageID::UPDATE_RATE:
+        if (_msg.payload_recv == 1) {
+            _sensor_state.update_rate = _msg.payload[0];
+        }
+        break;
+
+    case MessageID::DISTANCE_DATA_CM: {
+        // ignore distance messages until initialisation is complete
+        if (!_init_complete || (_msg.payload_recv != (PROXIMITY_SF45B_DESIRED_FIELD_COUNT * 2))) {
+            break;
+        }
+        _last_distance_received_ms = AP_HAL::millis();
+        const float distance_m = (int16_t)buff_to_uint16(_msg.payload[0], _msg.payload[1]) * 0.01f;
+        const float angle_deg = correct_angle_for_orientation((int16_t)buff_to_uint16(_msg.payload[2], _msg.payload[3]) * 0.01f);
+
+        // if distance is from a new sector then update distance, angle and boundary for previous sector
+        const uint8_t sector = convert_angle_to_sector(angle_deg);
+        if (sector != _sector) {
+            if (_sector != UINT8_MAX) {
+                _angle[_sector] = _sector_angle;
+                _distance[_sector] = _sector_distance;
+                _distance_valid[_sector] = _sector_distance_valid;
+                update_boundary_for_sector(_sector, false);
+            }
+            // init for new sector
+            _sector = sector;
+            _sector_angle = 0;
+            _sector_distance = INT16_MAX;
+            _sector_distance_valid = false;
+        }
+
+        // if distance is from a new minisector then update obstacle database using angle and distance from previous minisector
+        const uint8_t minisector = convert_angle_to_minisector(angle_deg);
+        if (minisector != _minisector) {
+            if ((_minisector != UINT8_MAX) && _minisector_distance_valid) {
+                database_push(_minisector_angle, _minisector_distance);
+            }
+            // init mini sector
+            _minisector = minisector;
+            _minisector_angle = 0;
+            _minisector_distance = INT16_MAX;
+            _minisector_distance_valid = false;
+        }
+
+        // check reading is valid
+        if (!ignore_reading(angle_deg) && (distance_m >= distance_min()) && (distance_m <= distance_max())) {
+            // update shortest distance for this sector
+            if (distance_m < _sector_distance) {
+                _sector_angle = angle_deg;
+                _sector_distance = distance_m;
+                _sector_distance_valid = true;
+            }
+
+            // update shortest distance for this mini sector
+            if (distance_m < _minisector_distance) {
+                _minisector_angle = angle_deg;
+                _minisector_distance = distance_m;
+                _minisector_distance_valid = true;
+            }
+        }
+        break;
+    }
+
+    default:
```

Comment:
Commenting unsupported messages out in the enumeration in the header would allow removal of `default`s like this one, and help when implementing a new message type.

Reply:
yes, although lots of commented out lines in the enum is pretty ugly.   I thought about this and originally did *not* have the default and instead had all the other unsupported messages listed, but this also seemed messy.